### PR TITLE
fix ajax fetchflowgraph request Nullpointexception

### PR DESCRIPTION
**Pull request by JasonBian**
fillFlowInfo method need to check Flow flow = project.getFlow(flowId); whether the flow is null or not.

### CODE REVIEW

**Comment by JasonBian** [comment]
add NullPointException check for the ajax request fetchflowgraph

**Comment by fsi206914** [comment]
Hi @JasonBian , could you please give us more details about this bug?

**Comment by JasonBian** [comment]
@fsi206914 when I use ajax api to invoke fetchflowgraph method, if I post invalid flowId, the fillFlowInfo method of the ProjectManagerServlet should check `Flow flow = project.getFlow(flowId)` is null or not.

File: azkaban-webserver/src/main/java/azkaban/webapp/servlet/ProjectManagerServlet.java [hunk 1]
```diff
@@ -999,7 +1004,7 @@ private void ajaxChangePermissions(Project project,
     boolean group = Boolean.parseBoolean(getParam(req, "group"));
 
     String name = getParam(req, "name");
-    Permission perm;
+    Permission perm;fillFlowInfo
```
Comment (HappyRay) [inline azkaban-webserver/src/main/java/azkaban/webapp/servlet/ProjectManagerServlet.java]: What's "fillFlowInfo" for?

File: azkaban-webserver/src/main/java/azkaban/webapp/servlet/ProjectManagerServlet.java [hunk 2]
```diff
@@ -790,37 +790,42 @@ private void fillFlowInfo(Project project, String flowId,
...
-      }
-
-      nodeList.add(nodeObj);
-      Set<Edge> inEdges = flow.getInEdges(node.getId());
-      if (inEdges != null && !inEdges.isEmpty()) {
-        ArrayList<String> inEdgesList = new ArrayList<String>();
-        for (Edge edge : inEdges) {
-          inEdgesList.add(edge.getSourceId());
+    if (flow != null) {
+      for (Node node : flow.getNodes()) {
```
Comment (HappyRay) [inline azkaban-webserver/src/main/java/azkaban/webapp/servlet/ProjectManagerServlet.java]: Consider breaking this logic out to its own method?

**Comment by HappyRay** [comment]
The logic LGTM. Thanks

**Pull request closed without merging**